specifyweb/specify/record_merging.py

```diff
@@ -197,13 +197,13 @@ def record_merge_fx(model_name: str, old_model_ids: List[int], new_model_id: int
         # Fix by optimizing the query by consolidating it here
         if model_name.lower() in MERGING_OPTIMIZATION_FIELDS and \
             table_name.lower() in MERGING_OPTIMIZATION_FIELDS[model_name.lower()]:
-            for field_name in MERGING_OPTIMIZATION_FIELDS[model_name.lower()][table_name.lower()]:
+            if field_name in MERGING_OPTIMIZATION_FIELDS[model_name.lower()][table_name.lower()]:
                 query = Q(**{field_name: old_model_ids[0]})
                 for old_model_id in old_model_ids[1:]:
                     query.add(Q(**{field_name: old_model_id}), Q.OR)
                 foreign_model.objects.filter(query).update(**{field_name: new_model_id})
                 progress(1, 0) if progress is not None else None
-            continue
+                continue
```

Comment:
`continue` is useless here as you are not in a `for` anymore

Reply:
ha ok, I added it back under @realVinayak advise

Reply:
because there is a for loop a couple of lines up!